articles/imagesharp/security.md

```diff
@@ -0,0 +1,19 @@
+# Security Considerations
+
+Image processing is a memory-intensive application. Most image processing libraries (including ImageSharp, SkiaSharp and Magick.NET) decode images into in-memory buffers for further processing. Without additional measures, any publicly facing service that consumes images coming from untrusted sources might be vulnerable to DoS attacks attempting to deplate process memory.
+
+Such measures can be:
+- Authentication, for example by using HMAC. See [Securing Processing Commands in ImageSharp.Web](../imagesharp.web/processingcommands.md#securing-processing-commands).
+- Offloading to separate services/containers.
+- Placing the solution behind a reverse proxy.
```

Comment:
I'm not sure I understand this one given that you already mentioned offloading to a separate service. Would using a reverse proxy just be a way to implement the former?

Great list otherwise :+1:

Reply:
In the offloading point I meant isolation of the image processing component, making sure that if it falls it doesn't bring the rest of the system with it.

Here I rather meant utlizing stuff like DDoS protection in cloudflare, but I must admit that I have no idea how that stuff works :)

Reply:
I see, I don't think cloudflare would really help in this particular case, but it's not a bad recommendation in general